### PR TITLE
Transfer Amounts (amount/estimated_amount)

### PR DESCRIPTION
**Pull request by nalancer08**
### Type of change

- [ ] Chore
- [X] Feature
- [ ] Bug fix

### Linked issue

https://cybrid.atlassian.net/browse/CYB-1358

### Why

We only display amount when the `transfer` is in state completed, in other case we display estimated_amount value.

### Evidence

https://github.com/Cybrid-app/cybrid-sdk-ios/assets/7268597/94f62e8e-766a-4c9f-af2e-dc43f1b94633

### CODE REVIEW

File: CybridSDK/Service Layer/Repositories/TransfersRepository.swift [hunk 1]
```diff
@@ -48,6 +48,6 @@ extension TransfersAPI: TransfersRepository {
     }
 
     static func fetchTransfers(accountGuid: String, _ completion: @escaping FetchTransfersCompletition) {
-        listTransfers(accountGuid: accountGuid, completion: completion)
+        listTransfers(perPage: 40, accountGuid: accountGuid, completion: completion)
```
Comment (brent-cybrid) [inline CybridSDK/Service Layer/Repositories/TransfersRepository.swift:51]: Why this change?

Comment (nalancer08) [reply CybridSDK/Service Layer/Repositories/TransfersRepository.swift:51]: I made some changes over the TableView -> Cell, I try to test something called reuseCellIdentifier that is when you scroll over a lot of items the cells keep states, for that reason I add this to test the scrolls and cell states works pretty fine

File: CybridSDK/Components/Accounts/Modal/AccountTransferModal.swift [hunk 2]
```diff
@@ -61,7 +60,7 @@ class AccountTransferModal: UIModal {
 
         // -- Sub Title View
         let amountCDecimal = CDecimal(transfer.amount ?? 0)
```
Comment (michael-cybrid) [inline CybridSDK/Components/Accounts/Modal/AccountTransferModal.swift]: Can this line be removed now? I don't see `amountCDecimal` being used anywhere.

File: CybridSDK/Components/Accounts/View/AccountTransfersViewModel.swift [hunk 3]
```diff
@@ -45,6 +45,20 @@ class AccountTransfersViewModel: NSObject {
             }
         }
     }
+
+    internal static func getAmountOfTransfer(_ transfer: TransferBankModel) -> String {
+
+        let amount = transfer.state == .completed ? transfer.amount ?? 0 : transfer.estimatedAmount ?? 0
+        let amountValue = CDecimal(amount)
+        let amountFormatted = AssetFormatter.forBase(Cybrid.fiat, amount: amountValue)
```
Comment (michael-cybrid) [inline CybridSDK/Components/Accounts/View/AccountTransfersViewModel.swift]: What is `Cybrid.fiat`? The transfer (in the case of book transfers) could be crypto (e.g. `BTC`).

File: CybridSDK/Components/Transfer/Modal/TransferModal.swift [hunk 4]
```diff
@@ -234,8 +234,8 @@ extension TransferModal {
         icon.asFirstInCenter(self.componentContent, size: UIValues.successIconSize, margins: UIValues.successIconMargins)
 
         // -- Value
-        let amountValueString = self.transferViewModel.currentTransfer.value?.amount
-        let amountBase = AssetFormatter.forBase(Cybrid.fiat, amount: CDecimal(amountValueString!))
+        let amountValueString = self.transferViewModel.currentTransfer.value?.estimatedAmount ?? 0
+        let amountBase = AssetFormatter.forBase(Cybrid.fiat, amount: CDecimal(amountValueString))
```
Comment (michael-cybrid) [inline CybridSDK/Components/Transfer/Modal/TransferModal.swift]: Should this go into a helper like `getAmountOfTransfer`?

File: CybridSDK/Components/Accounts/Modal/AccountTransferModal.swift (hunk 2 above)
Comment (nalancer08) [reply CybridSDK/Components/Accounts/Modal/AccountTransferModal.swift]: Is used in the next line

```
let amountCDecimal = CDecimal(transfer.amount ?? 0)
let amountFormatted = AssetFormatter.forBase(asset, amount: amountCDecimal)
```

File: CybridSDK/Components/Accounts/View/AccountTransfersViewModel.swift (hunk 3 above)
Comment (nalancer08) [reply CybridSDK/Components/Accounts/View/AccountTransfersViewModel.swift]: Is the fiat asset passed to the SDK init throw the bank configuration

```
let defaultAssetCode = self.bank?.supportedFiatAccountAssets?.first
self.fiat = assets.first(where: { $0.code == defaultAssetCode }) ?? FiatConfig.usd.defaultAsset
```

File: CybridSDK/Components/Transfer/Modal/TransferModal.swift (hunk 4 above)
Comment (nalancer08) [reply CybridSDK/Components/Transfer/Modal/TransferModal.swift]: Ready:

`let amountBaseFormatted = self.transferViewModel.getAmountOfCurrentTransferFormatted()`

And the function looks like:

```
func getAmountOfCurrentTransferFormatted() -> String {

        let amountValueString = self.currentTransfer.value?.amount
        let amountBase = AssetFormatter.forBase(Cybrid.fiat, amount: CDecimal(amountValueString!))
        return AssetFormatter.format(Cybrid.fiat, amount: amountBase)
}
```

File: CybridSDK/Components/Transfer/View/TransferViewModel.swift [hunk 5]
```diff
@@ -218,4 +218,11 @@ class TransferViewModel: NSObject {
         let side: PostQuoteBankModel.SideBankModel = self.isWithdraw.value ? .withdrawal : .deposit
         return side
     }
+
+    func getAmountOfCurrentTransferFormatted() -> String {
+
+        let amountValueString = self.currentTransfer.value?.amount
+        let amountBase = AssetFormatter.forBase(Cybrid.fiat, amount: CDecimal(amountValueString!))
+        return AssetFormatter.format(Cybrid.fiat, amount: amountBase)
```
Comment (michael-cybrid) [inline CybridSDK/Components/Transfer/View/TransferViewModel.swift]: Should this use `Cybrid.findAsset(...)` instead of `Cybrid.fiat`? Same with line 225.

Comment (nalancer08) [reply CybridSDK/Components/Transfer/View/TransferViewModel.swift]: Done!

**Review by michael-cybrid — APPROVED** [review]
